pact-jvm-consumer-kotlin/README.adoc

```diff
@@ -0,0 +1,156 @@
+  (from `PactDslWithProvider` to `RequestResponsePact`)
+* Support both:
+  1. prose like infix syntax (e.g. `"consumer" and "provider" havePact`)
+  2. fluent interface syntax (e.g. `consumer("consumer").hasPactWith("provider")`)
+
+== Examples
+
+[source,java,indent=0,role="primary"]
+.Infix DSL
+----
+include::src/test/kotlin/au/com/dius/pact/consumer/kotlin/dsl/examples/InfixDslExample.kt[lines=11..47]
+----
+
+[source,java,indent=0,role="secondary"]
+.Fluent DSL
+----
+include::src/test/kotlin/au/com/dius/pact/consumer/kotlin/dsl/examples/FluentDslExample.kt[lines=11..47]
+
+----
+
+[source,java,indent=0,role="secondary"]
+.ConsumerPactBuilder
+----
+include::src/test/kotlin/au/com/dius/pact/consumer/kotlin/dsl/examples/ConsumerPactBuilderExample.kt[lines=7..43]
+----
+
+== Usage together with https://github.com/DiUS/pact-jvm/tree/master/pact-jvm-consumer-junit5[pact-jvm-consumer-junit5]
+
+[source,java,indent=0,role="primary"]
+.kPact
+----
+include::src/test/kotlin/au/com/dius/pact/consumer/kotlin/dsl/examples/Junit5Example.kt[lines=14..26]
+----
+
+[source,java,indent=0,role="secondary"]
+.ConsumerPactBuilder
+----
+include::src/test/kotlin/au/com/dius/pact/consumer/kotlin/dsl/examples/Junit5Example.kt[lines=30..41]
+----
+
+== Open Design Decisions
+
+=== kPact block VS enforcing single Consumer/Provider pair
+
+==== Examples
+
+[source,java,indent=0,role="primary"]
+.kpact block example
+----
+kPact {
+    consumer withName "consumer" andProvider "provider" havePact {
+        // ...
+    }
+}
+----
+
+[source,java,indent=0,role="secondary"]
+.enforcing single Consumer/Provider pair example
+----
+KPact between "consumer" andProvider "provider" isDefinedBy {
+    // ...
+}
+----
+
+==== (Dis-)Advantages
+
+===== kpact block
+
+{n}
+  **compile time safety** / unexpected and inconsistent behavior: +
+  Multiple consumer/provider blocks are not prevented by compiler but
+  only last one would be used:
+[source,java,indent=0]
+----
+      kPact {
+          consumer withName "consumer" andProvider "provider" havePact { // <1>
+              // ...
+          }
+
+          consumer withName "consumer" andProvider "provider" havePact {
+              // ...
+          }
+      }
+----
+<1> This `consumer` block is not used at all
+
+
+{p}
+  Indicates clearer, that the block defines a (k)Pact
+
+===== enforcing single pair
+
+{p}
+  one indentation level less
+
+=== String extension function
+
+==== Examples
+
+===== without String extension
+
+[source,java,indent=0,role="primary"]
+.infix
+----
+consumer withName "consumer" andProvider "provider" havePact {
+----
+
+[source,java,indent=0,role="secondary"]
+.fluent
+----
+consumer("consumer").hasPactWith("provider") {
+----
+
+===== with String extension
+
+[source,java,indent=0,role="primary"]
+.infix
+----
+"consumer" and "provider" havePact {
+----
+[source,java,indent=0,role="secondary"]
+.fluent
+----
+"consumer".hasPactWith("provider") {
```

Comment:
I feel like a string extension goes too far. A string does not represent a consumer by itself. So "consumer".hasPactWith reads kind of non-sensical to me